File: createCards/artist_spread.m
clear all; close all; clc

folder_path = "mixed_v0";
file_name = "hitster_data_mixed_v0.xlsx";
column_name = "Artist";

% Construct the full file path
file_path = fullfile(folder_path, file_name);

% Read the data from the specified column in the Excel file
data = readtable(file_path);
artist_str = data{:, column_name};

% Convert artist_str to a categorical array
artist_categorical = categorical(artist_str);

% Count the occurrences of each artist
artist_counts = tabulate(artist_categorical);

% Sort the artists by the number of songs in descending order
[~, sortIdx] = sort([artist_counts{:, 2}], 'descend');
sorted_artists = artist_counts(sortIdx, 1);
sorted_counts = [artist_counts{sortIdx, 2}];

% Display the top 3 artists with the most songs
top_n = 5;
if numel(sorted_artists) < top_n
    top_n = numel(sorted_artists); % Adjust if fewer than 3 artists
end

fprintf('Top %d Artists with the Most Songs:\n', top_n);
for i = 1:top_n
    fprintf('%s: %d songs\n', sorted_artists{i}, sorted_counts(i));
end

File: createCards/artist_spread.py
import pandas as pd
import os

# Configuration
folder_path = "hitster_v2"
file_name = "hitster_data_hitster_v2.xlsx"
column_name = "Artist"  # Artist or SongName

# Construct the full file path
file_path = os.path.join(folder_path, file_name)

# Read the data from the specified column in the Excel file
data = pd.read_excel(file_path)
artist_data = data[column_name]

# Count the occurrences of each artist
artist_counts = artist_data.value_counts()

# Display the top artists with the most songs
top_n = 30
if len(artist_counts) < top_n:
    top_n = len(artist_counts)  # Adjust if fewer than 30 artists

print(f'Top {top_n} Artists with the Most Songs:')

# Convert to DataFrame for easier sorting
top_artists_df = pd.DataFrame({
    'Artist': artist_counts.head(top_n).index,
    'Count': artist_counts.head(top_n).values
})

# Sort by count (descending) then by artist name (alphabetically)
top_artists_df = top_artists_df.sort_values(['Count', 'Artist'], ascending=[False, True])

for _, row in top_artists_df.iterrows():
    print(f'{row["Artist"]}: {row["Count"]} songs')